packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -94,7 +95,7 @@ export class CheckboxBase extends BaseComponent<ICheckboxProps, ICheckboxState>
             disabled={disabled}
             ref={this._checkBox}
             name={name}
-            id={this._id}
+            id={id ? id : this._id}
```

Comment:
pretty sure this._id is used lateron down in the component. Can we just set this._id to either the generated value, or the passed in one? That way we don't have to keep checking for ID when we want the default ID value.